src/trace/ord/merge_batcher/mod.rs

```diff
@@ -172,22 +187,30 @@ impl<D: Ord, R: MonoidValue> MergeSorter<D, R> {
 
             // Consolidate and push the batch we were given
             consolidation::consolidate(&mut batch);
-            self.queue.push(vec![batch]);
-
-            // While there's at least two elements in our queue and one
-            // of them is much larger than the other
-            while self.queue.len() > 1
-                && (self.queue[self.queue.len() - 1].len()
-                    >= self.queue[self.queue.len() - 2].len() / 2)
-            {
-                let (left, right) = (
-                    self.queue.pop().expect("there's at least two batches"),
-                    self.queue.pop().expect("there's at least two batches"),
-                );
-
-                // Merge the two batches together
-                let merged = self.merge_by(left, right);
-                self.queue.push(merged);
+            if !batch.is_empty() {
+                if let Some(last) = self.queue.last_mut() {
+                    last.push(batch);
```

Comment:
This violates the invariant that the entire list of batches must be ordered, not just individual batches. This is why the original code pushes a single batch: `vec![batch]`